tests/test_core.py

```diff
@@ -941,6 +948,32 @@ def test_is_allowed_path(self):
             with pytest.raises(AssertionError):
                 self.config.is_allowed_path(None)
 
+    def test_event_on_update(self):
```

Comment:
let's not add new tests that are connected to a class.

Reply:
Fixed.